critiquebrainz/ws/review/views.py

```diff
@@ -384,10 +391,27 @@ def review_list_handler():
             language=language,
         )
         reviews = [db_review.to_dict(p) for p in reviews]
-        cache.set(cache_key, {
-            'reviews': reviews,
-            'count': count,
-        }, namespace=REVIEW_CACHE_NAMESPACE, time=REVIEW_CACHE_TIMEOUT)
+
+        if entity_id:
+            cache.set(cache_key, {
+                'reviews': reviews,
+                'count': count,
+            }, namespace=REVIEW_CACHE_NAMESPACE, time=REVIEW_CACHE_TIMEOUT)
+
+            # When reviews for an entity get created/updated, the existing cache for it need to be cleared. The keys
+            # used to cache reviews include parameters like user_id, sort, limit so on which are unavailable later.
+            # Hence, making it impossible to generate the cache keys at time of invalidation. To circumvent this issue,
+            # we track all cache keys for an entity_id. Cache keys of the form ws_cache_{entity_id} point to a list of
+            # cache keys of that entity. We can retrieve this list at time during invalidate and invalidate all caches
+            # for that entity_id.
```

Comment:
It's good to add this comment, but I think we can improve it a bit.

When we cache the results of a request, we include (entity_id, user_id, sort, limit, offset, language) in the cache key.
When entity_id is edited, we need to expire all cache items for this entity.
To do this, we track all of the cache keys for the entity in a separate cache item, ws_cache_{entity_id}.
These keys are retrieved and all keys are expired in `invalidate_ws_entity_cache`.